common/exception/ErrorMessage.java

```diff
@@ -414,12 +414,24 @@ public static void loadConstants() {
                 new TypeWrite(36, "The ability of type '%s' to own attribute type '%s' cannot be undefined because it is currently owned by existing instances.");
         public static final TypeWrite INVALID_UNDEFINE_PLAYS_HAS_INSTANCES =
                 new TypeWrite(37, "The ability of type '%s' to play role type '%s' cannot be undefined because it is currently played by existing instances.");
+        public static final TypeWrite INVALID_UNDEFINE_INHERITED_RELATES =
+                new TypeWrite(38, "The ability of type '%s' to relate role type '%s' cannot be undefined because the ability is inherited from a supertype.");
+        public static final TypeWrite INVALID_UNDEFINE_INHERITED_OWNS =
+                new TypeWrite(39, "The ability of type '%s' to own attribute type '%s' cannot be undefined because the ability is inherited from a supertype.");
+        public static final TypeWrite INVALID_UNDEFINE_INHERITED_PLAYS =
+                new TypeWrite(40, "The ability of type '%s' to play role type '%s' cannot be undefined because the ability is inherited from a supertype.");
```

Comment:
Major change of this request is making attempting to undefine nonexistent graph edges throw instead of no-op. This necessitated these new exceptions for attempting to undefine inherited relate/own/play edges...